vertical-pod-autoscaler/pkg/recommender/logic/recommender.go

```diff
@@ -108,13 +116,22 @@ func FilterControlledResources(estimation model.Resources, controlledResources [
 
 // CreatePodResourceRecommender returns the primary recommender.
 func CreatePodResourceRecommender() PodResourceRecommender {
-	targetEstimator := NewPercentileEstimator(*targetCPUPercentile, *targetMemoryPercentile)
-	lowerBoundEstimator := NewPercentileEstimator(*lowerBoundCPUPercentile, *lowerBoundMemoryPercentile)
-	upperBoundEstimator := NewPercentileEstimator(*upperBoundCPUPercentile, *upperBoundMemoryPercentile)
-
-	targetEstimator = WithMargin(*safetyMarginFraction, targetEstimator)
-	lowerBoundEstimator = WithMargin(*safetyMarginFraction, lowerBoundEstimator)
-	upperBoundEstimator = WithMargin(*safetyMarginFraction, upperBoundEstimator)
+	targetCPU := NewPercentileCPUEstimator(*targetCPUPercentile)
+	lowerBoundCPU := NewPercentileCPUEstimator(*lowerBoundCPUPercentile)
+	upperBoundCPU := NewPercentileCPUEstimator(*upperBoundCPUPercentile)
+
+	// Create base memory estimators
+	targetMemory := NewPercentileMemoryEstimator(*targetMemoryPercentile)
+	lowerBoundMemory := NewPercentileMemoryEstimator(*lowerBoundMemoryPercentile)
+	upperBoundMemory := NewPercentileMemoryEstimator(*upperBoundMemoryPercentile)
+
+	// Apply safety margins
+	targetCPU = WithCPUMargin(*safetyMarginFraction, targetCPU)
+	targetMemory = WithMemoryMargin(*safetyMarginFraction, targetMemory)
+	lowerBoundCPU = WithCPUMargin(*safetyMarginFraction, lowerBoundCPU)
+	lowerBoundMemory = WithMemoryMargin(*safetyMarginFraction, lowerBoundMemory)
+	upperBoundCPU = WithCPUMargin(*safetyMarginFraction, upperBoundCPU)
+	upperBoundMemory = WithMemoryMargin(*safetyMarginFraction, upperBoundMemory)
```

Comment:
This is potentially a nitpick...
```suggestion
	targetCPU = WithCPUMargin(*safetyMarginFraction, targetCPU)
	lowerBoundCPU = WithCPUMargin(*safetyMarginFraction, lowerBoundCPU)
	upperBoundCPU = WithCPUMargin(*safetyMarginFraction, upperBoundCPU)
	
	targetMemory = WithMemoryMargin(*safetyMarginFraction, targetMemory)
	lowerBoundMemory = WithMemoryMargin(*safetyMarginFraction, lowerBoundMemory)
	upperBoundMemory = WithMemoryMargin(*safetyMarginFraction, upperBoundMemory)
```

Just to bundle each resource together, much like the few lines above.
My brain seems to prefer it this way, what do you think?

Reply:
Yes, I agree. it’s more readable that way.